### PR TITLE
DDCE-2706-Mongo Migration

### PR DESCRIPTION
**Pull request by anu9**

### CODE REVIEW

File: app/repositories/SessionRepository.scala [hunk 1]
```diff
@@ -17,116 +17,81 @@
...
+@ImplementedBy(classOf[DefaultSessionRepository])
 trait SessionRepository {
 
   def get(id: String): Future[Option[UserAnswers]]
 
   def set(userAnswers: UserAnswers): Future[Boolean]
 
-  def resetCache(internalId: String): Future[Option[JsObject]]
-}
+  def resetCache(internalId: String): Future[Option[Boolean]]
```
Comment (frikit) [inline app/repositories/SessionRepository.scala:95]: why you have Option[Boolean] and not just boolean? you can default it to false if nothing got back from database like here:
```
collection.updateOne(selector, modifier, updateOptions).headOption().map(_.exists(_.wasAcknowledged()))
```
this is just an example how I did in other project your return type might be different so might need some changes

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -15,6 +15,7 @@ object AppDependencies {
 
   val test: Seq[ModuleID] = Seq(
     "org.scalatest"               %% "scalatest"                % "3.2.12",
+    "uk.gov.hmrc.mongo"           %% "hmrc-mongo-test-play-28"  %"0.73.0",
```
Comment (frikit) [inline project/AppDependencies.scala:18]: I dont see you using anything from this test library if you are not using it please delete it or if you want to add it here please add tests for repository which is better

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -15,6 +15,7 @@ object AppDependencies {
 
   val test: Seq[ModuleID] = Seq(
     "org.scalatest"               %% "scalatest"                % "3.2.12",
```
Comment (frikit) [inline project/AppDependencies.scala:17]: can be updated

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -5,7 +5,7 @@ object AppDependencies {
 
   val compile = Seq(
     play.sbt.PlayImport.ws,
-    "org.reactivemongo" %% "play2-reactivemongo"            % "0.20.13-play28",
+    "uk.gov.hmrc.mongo" %%   "hmrc-mongo-play-28"           % "0.73.0",
     "uk.gov.hmrc"       %% "play-frontend-hmrc"             % "1.1.0-play-28",
     "uk.gov.hmrc"       %% "play-conditional-form-mapping"  % "1.9.0-play-28",
```
Comment (frikit) [inline project/AppDependencies.scala:10]: can be updated

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -5,7 +5,7 @@ object AppDependencies {
...
     "uk.gov.hmrc"       %% "bootstrap-frontend-play-28"     % "5.24.0",
```
Comment (frikit) [inline project/AppDependencies.scala:11]: can be updated

File: project/AppDependencies.scala [hunk 6]
```diff
@@ -15,6 +15,7 @@ object AppDependencies {
 
   val test: Seq[ModuleID] = Seq(
     "org.scalatest"               %% "scalatest"                % "3.2.12",
+    "uk.gov.hmrc.mongo"           %% "hmrc-mongo-test-play-28"  %"0.73.0",
     "org.scalatestplus.play"      %% "scalatestplus-play"       % "5.1.0",
     "org.pegdown"                 %  "pegdown"                  % "1.6.0",
```
Comment (frikit) [inline project/AppDependencies.scala:20]: should be replaced with flexmark

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -15,6 +15,7 @@ object AppDependencies {
...
     "org.jsoup"                   %  "jsoup"                    % "1.15.1",
```
Comment (frikit) [inline project/AppDependencies.scala:21]: all dependencies here can be updated, I suggest to delete all mockito one and add scalatest-plus.mockito in here only, for scalacheck is the same

Also dont need overrides if you update all libs here

**Review by frikit — CHANGES_REQUESTED** [review]
- Could you please add run_all_test script like here https://github.com/hmrc/estates-store/pull/22/files#diff-d1d11ed87c55a802e257d37d5180a6dfbc114d1c0686f8db63ef83d6d94ab2bf to run all test and all check much easier
- Could you please add depency-updates plugins for easier maintain and updating libraries in future https://github.com/hmrc/estates-store/pull/22/files#diff-8fd74420ca55aa91d4c53775d2a55a999c405b5a4fafeaf73b7f0ef83e1f3b0c
- Remove repositories from ScoverageKeys.coverageExcludedFiles , as we should add tests for that package not ignore it
- after that your coverage might go lower so we need add more tests to bring it back
- plugins can be updated
- silencer plugin can be updated
- Please fix scala compiler warnings
- Please fix unused imports in DefaultSessionRepository
- Please fix warnings in scalastyle.xml
- Please fix some scalastyle warnings
- Please fix warning in build.sbt at compile

**Comment by frikit** [comment]
Please also fix commit message with the pattern JIRA_NR - Commit message what changes have you done or [JIRA_NR] - ...

File: app/repositories/IndexManager.scala [hunk 8]
```diff
@@ -14,61 +14,77 @@
  * limitations under the License.
  */
 
-package repositories
```
[the remark this answers is not in the document]
Comment (anu9) [reply app/repositories/IndexManager.scala:17]: Deleted the file

File: app/repositories/SessionRepository.scala [hunk 9]
```diff
@@ -17,116 +17,81 @@
...
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions().name("user-answers-last-updated-index").expireAfter(config.cachettlInSeconds, TimeUnit.SECONDS).unique(false)),
+      IndexModel(
+        Indexes.ascending("internalId"),
+        IndexOptions().name("internal-auth-id-index")
+      )
+    )
   )
+    with Logging with SessionRepository {
```
[the remark this answers is not in the document]
Comment (anu9) [reply app/repositories/SessionRepository.scala]: Deleted

File: project/AppDependencies.scala (hunk 4 above)
Comment (anu9) [reply project/AppDependencies.scala]: done

File: project/AppDependencies.scala (hunk 5 above)
Comment (anu9) [reply project/AppDependencies.scala:11]: done

File: app/repositories/SessionRepository.scala [hunk 10]
```diff
@@ -17,116 +17,81 @@
...
+    domainFormat = Format(UserAnswers.reads,UserAnswers.writes),
+    collectionName = "user-answers",
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions().name("user-answers-last-updated-index").expireAfter(config.cachettlInSeconds, TimeUnit.SECONDS).unique(false)),
+      IndexModel(
+        Indexes.ascending("internalId"),
+        IndexOptions().name("internal-auth-id-index")
+      )
```
[the remark this answers is not in the document]
Comment (anu9) [reply app/repositories/SessionRepository.scala:51]: No DropIndexes val is in main branch thats why I didn't add here

File: project/AppDependencies.scala (hunk 3 above)
Comment (anu9) [reply project/AppDependencies.scala:17]: done

File: app/repositories/SessionRepository.scala (hunk 10 above)
[the remark this answers is not in the document]
Comment (frikit) [reply app/repositories/SessionRepository.scala:51]: please add replaceIndexes with that property back here to have the same logic implement in new library

File: project/AppDependencies.scala (hunk 6 above)
Comment (anu9) [reply project/AppDependencies.scala]: done

File: app/repositories/SessionRepository.scala (hunk 10 above)
[the remark this answers is not in the document]
Comment (anu9) [reply app/repositories/SessionRepository.scala:51]: ok done

File: project/AppDependencies.scala (hunk 7 above)
Comment (anu9) [reply project/AppDependencies.scala]: done

File: project/AppDependencies.scala (hunk 5 above)
Comment (frikit) [reply project/AppDependencies.scala:11]: this change is still not done

File: app/repositories/SessionRepository.scala (hunk 1 above)
Comment (frikit) [reply app/repositories/SessionRepository.scala:94]: this change is still not done

File: project/AppDependencies.scala [hunk 11]
```diff
@@ -5,25 +5,23 @@ object AppDependencies {
...
   val compile = Seq(
     play.sbt.PlayImport.ws,
-    "org.reactivemongo" %% "play2-reactivemongo"            % "0.20.13-play28",
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-play-28"             % "0.73.0",
     "uk.gov.hmrc"       %% "play-frontend-hmrc"             % "1.1.0-play-28",
-    "uk.gov.hmrc"       %% "play-conditional-form-mapping"  % "1.9.0-play-28",
+    "uk.gov.hmrc"       %% "play-conditional-form-mapping"  % "1.12.0-play-28",
     "uk.gov.hmrc"       %% "bootstrap-frontend-play-28"     % "5.24.0",
-    "com.typesafe.play" %% "play-json-joda"                 % "2.7.4",
+    "com.typesafe.play" %% "play-json-joda"                 % "2.9.3",
```
Comment (frikit) [inline project/AppDependencies.scala:12]: I dont see it used anywhere in project can be deleted

File: project/AppDependencies.scala [hunk 12]
```diff
@@ -5,25 +5,23 @@ object AppDependencies {
...
     "uk.gov.hmrc"       %% "domain"                         % "6.2.0-play-28"
```
Comment (frikit) [inline project/AppDependencies.scala:13]: can be updated

File: project/AppDependencies.scala [hunk 13]
```diff
@@ -43,4 +41,4 @@ object AppDependencies {
     "commons-codec"     %  "commons-codec"        % "1.12"
```
Comment (frikit) [inline project/AppDependencies.scala:41]: still need to delete overrides

**Review by frikit — CHANGES_REQUESTED** [review]
Still not done:
- Please fix some scalastyle warnings

File: project/AppDependencies.scala (hunk 11 above)
Comment (anu9) [reply project/AppDependencies.scala:12]: done

File: project/AppDependencies.scala (hunk 12 above)
Comment (anu9) [reply project/AppDependencies.scala:12]: not used deleted

File: app/repositories/SessionRepository.scala (hunk 1 above)
Comment (anu9) [reply app/repositories/SessionRepository.scala]: done

**Pull request closed without merging**